docs/2_architecture/components/2_payment_service_contract.md

```diff
@@ -20,12 +20,9 @@ the [Aligned Service Manager](./3_service_manager_contract.md),
 including tokens for batch verification payment to the [Aggregator](./5_aggregator.md).
 
 Users can then withdraw extra funds deposited to the Batcher Payments smart contract,
-or leave them to fund future proofs.
+or leave them to fund future proofs. For this, to avoid DoS'ing the Batcher, Users must first call the contract's to `unlock` function, and then call `withdraw` at least 100 blocks later, to finish their withdraw process. This enables the Batcher to be sure the User will have funds to pay for his proofs when `createNewTask` is called.
```

Comment:
```suggestion
or leave them to fund future proofs. For this, to avoid DoS'ing the Batcher, Users must first call the contract's `unlock` function, and then call `withdraw` at least 100 blocks later, to finish their withdraw process. This enables the Batcher to be sure the User will have funds to pay for his proofs when `createNewTask` is called.
```